### PR TITLE
Support compiling {fmt} as named module

### PR DESCRIPTION
**Pull request by DanielaE**
Start by tagging official API for module export

### CODE REVIEW

File: include/fmt/color.h [hunk 1]
```diff
@@ -231,6 +235,7 @@ struct color_type {
 }  // namespace detail
 
 /** A text style consisting of foreground and background colors and emphasis. */
+FMT_MODULE_EXPORT
```
Comment (vitaut) [inline include/fmt/color.h:238]: Why not include this class in the `FMT_MODULE_EXPORT_BEGIN/FMT_MODULE_EXPORT_END` block below and remove `FMT_MODULE_EXPORT`?

File: include/fmt/printf.h [hunk 2]
```diff
@@ -194,10 +194,12 @@ FMT_DEPRECATED void printf(detail::buffer<Char>& buf,
 }
 using detail::vprintf;
 
+// already exported through "ostream.h" above
 template <typename Char>
 class basic_printf_parse_context : public basic_format_parse_context<Char> {
   using basic_format_parse_context<Char>::basic_format_parse_context;
 };
+// already exported through "ostream.h" above
```
Comment (vitaut) [inline include/fmt/printf.h:202]: Let's remove this comment, the one above can apply to both contexts.

**Review by vitaut — COMMENTED** [review]
Thank you for the PR. LGTM, just two minor comments inline.

File: include/fmt/color.h (hunk 1 above)
Comment (DanielaE) [reply include/fmt/color.h:238]: Yes, why not? I'll tell ya: staring at code for extended periods of time didn't prevent me from being silly.

**Pull request merged**